src/ArrayDeserializer.php

```diff
@@ -261,4 +269,33 @@ protected function isBuiltInTypeValid(string $type, mixed $value, string $path):
             default => throw new UnsupportedTypeException($path, $type),
         };
     }
+
+    /**
+     * @param mixed $value
+     * @param string $path
+     * @return mixed
+     * @throws InvalidEnumBackingException
+     * @throws UnsupportedTypeException
+     * @noinspection PhpDocMissingThrowsInspection
+     */
+    protected function parseEnum(mixed $value, string $path): mixed
+    {
+        /** @noinspection PhpUnhandledExceptionInspection - It has already been verified that the enum exists */
+        $reflectionEnum = new ReflectionEnum($this->class);
+
+        if (!$reflectionEnum->isBacked()) {
```

Comment:
Couldn't unit enums be supported by using their index in the cases or their name?